cmd/agent/subcommands/dogstatsd/command.go

```diff
@@ -51,7 +51,7 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			return fxutil.OneShot(topContexts,
 				fx.Supply(&topFlags),
 				fx.Supply(core.BundleParams{
-					ConfigParams: cconfig.NewAgentParams(globalParams.ConfFilePath, cconfig.WithExtraConfFiles(globalParams.ExtraConfFilePath)),
+					ConfigParams: cconfig.NewAgentParams(globalParams.ConfFilePath, cconfig.WithExtraConfFiles(globalParams.ExtraConfFilePath), cconfig.WithFleetPoliciesDirPath(globalParams.FleetPoliciesDirPath)),
```

Comment:
Would it make sense to extract this as a helper function? This complex expression repeats twice in this file alone, and there seems a lot more of other uses.

Reply:
I agree that it's not the easiest thing to read, but I don't think it's a good thing to have a helper function here.

As the helper function simply lets us remove the `cconfig.With...` chars (we need to keep the paths as args), I'd say that it's not worth losing the options pattern flexibility (optional params, descriptive methods, testability...).

Reply:
I'm not concerned with readability, but with maintainability, specifically maintaining uniform behavior across numerous agent subcommands. 

> (we need to keep the paths as args)

We don't, the helper can accept a single `globalParams` value. If we have a shared definition for parameters, why their handling must be copy-pasted?

> it's not worth losing the options pattern flexibility

If all options specified are always the same, there is no gain from this flexibility, only extra work. And we don't need to remove the options pattern, just encapsulate one common use case.

Reply:
> we should have a common boilerplate for agent commands that'd include config management.

I'm definitely not proposing doing anything large in this PR, only to avoid copy-pasted  code where possible. Not not a blocker issue, so feel free to proceed as you see fit.